docs/pages/repo/docs/core-concepts/caching/file-inputs.mdx

```diff
@@ -0,0 +1,116 @@
+---
+title: File Inputs
+description: How to configure Turborepo file inputs
+---
+
+import Callout from "../../../../../components/Callout";
+
+# File Inputs
+
+By default, a workspace is considered to have been updated when any of the files in that workspace have changed. However, not every file change inside of the entire repo should result in rerunning a task. Turborepo gives you the tools to make sure that you can define every file that matters.
+
+## Global Files
+
+Almost every repository has some configuration that appears at the root of the repository. These files have the potential to impact the behavior of child workspaces.
+
+### `package.json` and `turbo.json`
+
+By default, Turborepo includes the root `package.json`, any lockfile that may be present, and the root `turbo.json` file into the global hash. You do not need to specify them separately.
+
+### Specifying Additional Inputs
+
+If there are additional files that should be considered for every single workspace, such as a root `tsconfig.json` you should specify that in `globalDependencies`:
+
+```jsonc filename="/turbo.json"
+{
+  "$schema": "https://turbo.build/schema.json",
+  "globalDependencies": ["tsconfig.json"],
+  "pipeline": {
+    "type-check": {}
+  }
+}
+```
+
+<Callout type="info">
+  This customization is _additive_ to Turborepo's default behavior for `package.json` and `turbo.json`.
+</Callout>
+
+In this example every single task in the turborepo would be rerun when the root `tsconfig.json` file changes.
+
+## Workspace Files
+
+In addition to global files, which appear outside of any workspace, you may also specify which files to include on a per-workspace basis.
+
+### `package.json` and `turbo.json`
+
+The contents of `package.json` and `turbo.json` are automatically considered for workspace hashes. This behavior is not configurable.
+
+### Turborepo's Default Behavior
+
+The default behavior of Turborepo is to inspect the repository and identify which files should be included in the hash calculation. Turborepo typically uses Git to identify which files to consider:
+
+- All of the files which appear in the Git index which are children of the workspace directory.
+- All of the untracked files which are children of the workspace directory and are not `.gitignore`d.
```

Comment:
I think "git index" and references to tracking is a little too deep into the `git` weeds. Maybe we can say anything not ignored by `git`?

Reply:
I intentionally made that distinction because the `.git/index` and `.gitignore` are functionally distinct. The index can contain arbitrary things that `.gitignore` would by default exclude.

In other words: I chose "accuracy" + "precision" here.